framework/include/restart/Restartable.h

```diff
@@ -87,6 +101,21 @@ class Restartable
   T &
   declareRestartableDataWithContext(const std::string & data_name, void * context, Args &&... args);
 
+  /**
+   * Declare a piece of data as "restartable" and initialize it.
+   * Similar to `declareRestartableDataWithContext` but returns a const reference to the object.
+   *
+   * NOTE: This returns a _ const reference_!  Make sure you store it in a _const reference_!
+   *
+   * @param data_name The name of the data (usually just use the same name as the member variable)
+   * @param context Context pointer that will be passed to the load and store functions
+   * @param args Arguments to forward to the constructor of the data
+   */
+  template <typename T, typename... Args>
+  const T & getRestartableDataWithContext(const std::string & data_name,
```

Comment:
Does your use case need context @grmnptr? As this is a very advanced cased (where you need to get something from a whole different object, with the knowledge of how it was declared previously), I would prefer to keep the interface locked down for now and only allow this for getting this _without_ context until the need arrives.

Reply:
Makes sense